ipn/ipnserver/server.go

```diff
@@ -1029,13 +1029,16 @@ func (s *Server) localhostHandler(ci connIdentity) http.Handler {
 // Windows and via $DEBUG_LISTENER/debug/ipn when tailscaled's --debug flag
 // is used to run a debug server.
 func (s *Server) ServeHTMLStatus(w http.ResponseWriter, r *http.Request) {
-	// As this is only meant for debug, verify there's no DNS name being used to
-	// access this.
-	if strings.IndexFunc(r.Host, unicode.IsLetter) != -1 {
+	// As this is only meant for debug, verify we are only serving to a
+	// localhost or equivalent (i.e. 127.0.0.1) origin.
+	if r.Host != "localhost" && strings.IndexFunc(r.Host, unicode.IsLetter) != -1 {
 		http.Error(w, "invalid host", http.StatusForbidden)
 		return
 	}
 
+	w.Header().Set("Content-Security-Policy", `default-src 'none'; frame-ancestors 'none'; script-src 'none'; script-src-elem 'none'; script-src-attr 'none'`)
```

Comment:
time to put all this stuff in tsweb or something?

starting to get repetitive

Reply:
Maybe as an option? We only want this set on stuff that must never be i-framed.

Reply:
oh I meant as helper funcs/options, yeah, not by default

Reply:
I'll leave this for another PR as discussed on slack